include/seqan3/search/algorithm/configuration/max_error_rate.hpp

```diff
@@ -63,47 +64,82 @@ struct search_config_max_error_rate
      */
-    template <typename configuration_t>
+    template <typename configuration_t, typename ... error_types_t>
     //!\cond
         requires is_algorithm_configuration_v<remove_cvref_t<configuration_t>>
     //!\endcond
-    // TODO: replace int by templates and check whether it can be casted to uint8_t
-    // TODO: template argument packing. allow any order and subset of strong types
-    constexpr auto invoke(configuration_t && cfg, search_cfg::total<double> const total_error,
-                                                  search_cfg::substitution<double> const substitution_error,
-                                                  search_cfg::insertion<double> const insertion_error,
-                                                  search_cfg::deletion<double> const deletion_error) const
+    constexpr auto invoke(configuration_t && cfg, error_types_t && ...error_types) const
     {
-        static_assert(is_valid_search_configuration_v<search_cfg::id::max_error, remove_cvref_t<configuration_t>>,
-                      SEQAN3_INVALID_CONFIG(search_cfg::id::max_error));
-
-        double total {total_error},
-               substitution {substitution_error},
-               insertion {insertion_error},
-               deletion {deletion_error};
-
-        if (0 > total        || total > 1 ||
-            0 > substitution || substitution > 1 ||
-            0 > insertion    || insertion > 1 ||
-            0 > deletion     || deletion > 1)
+        static_assert(is_valid_search_configuration_v<search_cfg::id::max_error_rate, remove_cvref_t<configuration_t>>,
+                      SEQAN3_INVALID_CONFIG(search_cfg::id::max_error_rate));
+
+        std::tuple in{error_types...};
+        std::tuple<search_cfg::total<double>, search_cfg::substitution<double>,
+                   search_cfg::insertion<double>, search_cfg::deletion<double>> out{.0, .0, .0, .0};
+
+        get_type_out<search_cfg::total<double>>(out, in);
+        get_type_out<search_cfg::substitution<double>>(out, in);
+        get_type_out<search_cfg::insertion<double>>(out, in);
+        get_type_out<search_cfg::deletion<double>>(out, in);
+
+        using error_types_list_t = meta::list<remove_cvref_t<error_types_t>...>;
+        constexpr bool total_set = meta::in<error_types_list_t, search_cfg::total<double>>::value;
+        constexpr bool other_error_types_set = meta::count<error_types_list_t, search_cfg::total<double>>::value !=
+                                               meta::size<error_types_list_t>::value;
+
+        if (0 > static_cast<double>(std::get<0>(out)) || static_cast<double>(std::get<0>(out)) > 1 ||
+            0 > static_cast<double>(std::get<1>(out)) || static_cast<double>(std::get<1>(out)) > 1 ||
+            0 > static_cast<double>(std::get<2>(out)) || static_cast<double>(std::get<2>(out)) > 1 ||
+            0 > static_cast<double>(std::get<3>(out)) || static_cast<double>(std::get<3>(out)) > 1)
```

Comment:
maybe extract the doubles once into variables and the re-use to increase readability